docs/guides/frontend-redux.md

```diff
@@ -0,0 +1,63 @@
+# Frontend Data Architecture
+
+This post describes the architecture used throughout [frontend-scaffold](https://github.com/Code-4-Community/frontend-scaffold), Code4Community's shared frontend template used by all client projects. The scaffold is a React application created with Create React App using the TypeScript template. It uses Redux for application state management. 
+
+## What is Redux? 
+
+Redux is a predictable state container for JavaScript apps.
+
+It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test. On top of that, it provides a great developer experience, such as live code editing combined with a time traveling debugger.
+
+Redux accomplishes this by establishing a set of core principles for accessing, fetching, and updating data in your application. A Redux store has one common *state* object, which you can think of as your "model", to use object-oriented terminology. This model has no "setters", to prevent different parts of the code from updating the data in different ways. 
```

Comment:
Why would you want to "prevent different parts of the code from updating the data in different ways?" What utility does this provide?

Reply:
Think of this as the equivalent of why we have private variables in Java classes with getters and setters to access them, rather than mutating those variables directly. It's a defined pattern of access and updating that makes debugging and understanding data flow much easier.